docs/source/markdown/podman-generate-spec.1.md

```diff
@@ -0,0 +1,29 @@
+% podman-generate-spec
+## NAME
+podman-generate-spec - Generate Specgen JSON based on containers or pods
+
+## SYNOPSIS
+**podman generate spec** [*options*] *container...* | *pod...*
+
+## DESCRIPTION
+**podman generate spec** will generate Specgen JSON from Podman Containers and Pods. This JSON can either be printed to a file, directly to the command line, or both.
+
+This JSON can then be used as input for the podman API, specifically for podman container and pod creation. Specgen is podman's internal structure for formulating new container related entities.
+
+## OPTIONS
+
+#### **--clone**
+
+Rename the pod or container so that it does not conflict with the existing entity. This is helpful when the JSON is to be used before the pod/container you are generating from is deleted.
+
+#### **--compact**, **-c**
+
+Print the output in a compact, one line format. This is useful when piping the data to the podman API
```

Comment:
Should we default to compact and force users to use | json_pp or equivalent?

Or if we generate YAML file by default, then --json would ouput in --compact format.

Reply:
@cdoern newlines are totally valid in json. You have an issue with your shell since you do not escape the output which is always a very bad idea. Try to put the `"$()"` in quotes.